backend/agentd/session.go

```diff
@@ -194,21 +202,22 @@ func (s *Session) Start() (err error) {
 	go s.subPump()
 
 	org, env := s.cfg.Organization, s.cfg.Environment
-	agentID := s.cfg.AgentID
 
 	defer func() {
 		if err != nil {
 			s.Stop()
 		}
 	}()
 
-	for _, sub := range s.cfg.Subscriptions {
+	for i, sub := range s.cfg.Subscriptions {
 		topic := messaging.SubscriptionTopic(org, env, sub)
 		logger.Debugf("Subscribing to topic %q", topic)
-		if err := s.bus.Subscribe(topic, agentID, s.checkChannel); err != nil {
+		subscription, err := s.bus.Subscribe(topic, s)
+		if err != nil {
 			logger.WithError(err).Error("error starting subscription")
 			return err
 		}
+		s.subscriptions[i] = subscription
```

Comment:
Heh, looks like the CI found a race condition here... seemingly with the Stop method? :man_shrugging:

Reply:
But only on 386? I didn't even think the race detector worked on anything but amd64

Reply:
Oh this is annoying. We have the exact same problem anywhere we have Start() and Stop(). Sessions and Daemons aren't safe for concurrent use, but neither of them is used concurrently.